addons/mail/static/src/composer/composer.js

```diff
@@ -128,10 +130,23 @@ export class Composer extends Component {
         useChildSubEnv({
             inComposer: true,
         });
-        useEmojiPicker(useRef("emoji-picker"), {
-            onSelect: (str) => this.addEmoji(str),
-            onClose: () => this.state.autofocus++,
-        });
+        if (!this.env.isSmall || !this.env.inChatWindow) {
+            usePopoverEmojiPicker(useRef("emoji-picker"), {
+                onSelect: (str) => this.addEmoji(str),
+                onClose: () => this.state.autofocus++,
+            });
+        } else {
+            this.storeScrollEmojiPicker = {
+                scrollValue: 0,
+                set: (value) => {
+                    this.storeScrollEmojiPicker.scrollValue = value;
+                },
+                get: () => {
+                    return this.storeScrollEmojiPicker.scrollValue;
+                },
+            };
+        }
+        onExternalClick("composer", () => (this.state.showEmojiPicker = false));
```

Comment:
As we discussed, it's no good to have 2 ways to use EmojiPicker (either hook or as sub-component in template).
This leads to lots of code duplication, and also it makes `composer` more complex than it should.

I would keep a single way to `useEmojiPicker` with the hook, and let the hook allow to use emoji picker with popover or non-popover based on params you pass to the hook.